include/deal.II/lac/solver_gmres.h

```diff
@@ -1773,10 +1775,20 @@ SolverFGMRES<VectorType>::solve(const MatrixType &        A,
           A.vmult(*aux, z[j]);
 
           // Gram-Schmidt
-          H(0, j) = *aux * v[0];
-          for (unsigned int i = 1; i <= j; ++i)
-            H(i, j) = aux->add_and_dot(-H(i - 1, j), v[i - 1], v[i]);
-          H(j + 1, j) = a = std::sqrt(aux->add_and_dot(-H(j, j), v[j], *aux));
+          bool         re_orthogonalize = false;
+          const double s =
+            internal::SolverGMRESImplementation::iterated_gram_schmidt<
+              VectorType>(SolverGMRES<VectorType>::AdditionalData::
+                            OrthogonalizationStrategy::modified_gram_schmidt,
```

Comment:
@kronbichler Could you try this out by manually setting this line to: `classical_gram_schmidt` (the test `lac/solver_02.debug` has passed)? In a follow-up PR, we should introduce paramters.

Reply:
I checked and it seems to work correctly, thank you.

Reply:
Thank your for testing! Once merged, we need to add a new parameter to `SolverFGMRES::AdditionalData`. What is your opinion of `SolverGMRES<VectorType>::AdditionalData::OrthogonalizationStrategy`. `SolverGMRES` is not the right place, is it. What about having `OrthogonalizationStrategy` directly in the `dealii` namespace?

Reply:
I gave this some thought, but my (subjective) feeling is that a struct `OrthogonalizationStrategy` in the deal.II namespace does not look very desirable. Since what we do here is a generic orthogonalization method (as part of an Arnoldi process), which also has other applications, I wonder if we should introduce a class in the `dealii` namespace for the overall `iterated_gram_schmidt` process? So a class that does not manage the memory except for the small matrix. Then, it would also be natural to let `SolverGMRES` and `SolverFGMRES` decide on `IteratedGramSchmidt<VectorType>::OrthogonalizationStrategy`. The reason why I'm thinking about a separate class is that I would also like to have a "projection" algorithm to solve a least squares problem (as initial guess to an iterative solver) in deal.II.
This does probably fit into a linear algebra namespace - inside that namespace, I would also not feel as opposed to keep the Gram-Schmidt algorithm in the internal namespace and expose only `LinearAlgebra::OrthogonalizationStrategy`. One could think about using the related QR factorization and embed the Gram-Schmidt process there, so we would then also have other means than Gram-Schmidt, such as Householder reflections (I did not think closely if anything else than Gram-Schmidt would be feasible for GMRES, probably not, so consider this with a grain of salt).